tests/test_security_gssapi.cpp

```diff
@@ -0,0 +1,303 @@
+
+const char *name = "zmqtest2";
+
+static volatile int zap_deny_all = 0;
+
+#ifdef ZMQ_BUILD_DRAFT_API
+//  Read one event off the monitor socket; return value and address
+//  by reference, if not null, and event number by value. Returns -1
+//  in case of error.
+
+static int
+get_monitor_event (void *monitor, int *value, char **address)
+{
+    //  First frame in message contains event number and value
+    zmq_msg_t msg;
+    zmq_msg_init (&msg);
+    if (zmq_msg_recv (&msg, monitor, 0) == -1)
+        return -1;              //  Interruped, presumably
+    assert (zmq_msg_more (&msg));
+
+    uint8_t *data = (uint8_t *) zmq_msg_data (&msg);
+    uint16_t event = *(uint16_t *) (data);
+    if (value)
+        *value = *(uint32_t *) (data + 2);
+
+    //  Second frame in message contains event address
+    zmq_msg_init (&msg);
+    if (zmq_msg_recv (&msg, monitor, 0) == -1)
+        return -1;              //  Interruped, presumably
+    assert (!zmq_msg_more (&msg));
+
+    if (address) {
+        uint8_t *data = (uint8_t *) zmq_msg_data (&msg);
+        size_t size = zmq_msg_size (&msg);
+        *address = (char *) malloc (size + 1);
+        memcpy (*address, data, size);
+        *address [size] = 0;
+    }
+    return event;
+}
+#endif
+
+//  --------------------------------------------------------------------------
+//  This methods receives and validates ZAP requestes (allowing or denying
+//  each client connection).
+//  N.B. on failure, each crypto type in keytab will be tried
+
+static void zap_handler (void *handler)
+{
+    //  Process ZAP requests forever
+    while (true) {
+        char *version = s_recv (handler);
+        if (!version)
+            break;          //  Terminating
+
+        char *sequence = s_recv (handler);
+        char *domain = s_recv (handler);
+        char *address = s_recv (handler);
+        char *identity = s_recv (handler);
+        char *mechanism = s_recv (handler);
+        char *principal = s_recv (handler);
+
+        assert (streq (version, "1.0"));
+        assert (streq (mechanism, "GSSAPI"));
+
+        s_sendmore (handler, version);
+        s_sendmore (handler, sequence);
+
+        if (!zap_deny_all) {
+            s_sendmore (handler, "200");
+            s_sendmore (handler, "OK");
+            s_sendmore (handler, "anonymous");
+            s_send     (handler, "");
+	    //fprintf (stderr, "ALLOW %s\n", principal);
+        }
+        else {
+            s_sendmore (handler, "400");
+            s_sendmore (handler, "Denied");
+            s_sendmore (handler, "");
+            s_send     (handler, "");
+	    //fprintf (stderr, "DENY %s\n", principal);
+        }
+        free (version);
+        free (sequence);
+        free (domain);
+        free (address);
+        free (identity);
+        free (mechanism);
+        free (principal);
+    }
+    zmq_close (handler);
+}
+
+int main (void)
+{
+#ifndef HAVE_LIBGSSAPI_KRB5
```

Comment:
Instead of having the IFDEF here it can be in the makefile.am directly, like other tests are doing, eg tipc. This way it's skipped and doesn't risk a false positive.